### PR TITLE
update TiDB roadmap for CY24-CY25

### PR DESCRIPTION
**Pull request by songrijie**
### What is changed, added or deleted? (Required)

Update the TiDB roadmap for the upcoming year.  

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-roadmap.md [hunk 1]
```diff
@@ -150,79 +265,64 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
           <li>
-            <b>列级访问控制</b>
-            <br />允许针对特定列来授予或限制访问权限
+            <b>完善动态权限</b>
+            <br />完善动态权限设计，限制 Super 实现。
           </li>
           <br />
+          <li>
+            <b>基于标记的日志脱敏</b>
+            <br />支持对集群日志中敏感数据进行标记，而后根据使用场景可以选择是否对敏感信息进行脱敏。
```
Comment (Oreoxmt) [inline tidb-roadmap.md:279]:
```suggestion
            <br />支持在集群日志中标记敏感数据，然后根据使用场景选择是否对这些敏感信息进行脱敏。
```

File: tidb-roadmap.md [hunk 2]
```diff
@@ -150,79 +265,64 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+            <br />支持多因素身份认证机制。
           </li>
         </ul>
       </td>
       <td>
         <ul>
           <li>
             <b>基于标签的访问控制</b>
-            <br />通过配置标签来授予访问权限
+            <br />支持通过配置标签的方式，通过标签形式对数据进行访问控制。
```
Comment (Oreoxmt) [inline tidb-roadmap.md:309]:
```suggestion
            <br />支持通过配置标签的方式，以标签形式对数据进行访问控制。
```

File: tidb-roadmap.md [hunk 3]
```diff
@@ -150,79 +265,64 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
           </li>
+          <br />
           <li>
             <b>增强客户端加密</b>
+            <br />支持客户端对关键字段加密，增强数据安全性
```
Comment (Oreoxmt) [inline tidb-roadmap.md:314]:
```suggestion
            <br />支持客户端对关键字段进行加密，增强数据安全性。
```

File: tidb-roadmap.md [hunk 4]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
       <td>
         <ul>
           <li>
-             <b>分布式并行执行框架</b><br />
-            TiDB v7.2.0 引入了用于后台任务（如 DDL 和 analyze）的分布式并行执行框架，为实现这些任务在计算节点间并行化提供了基础。v7.4.0 为分布式 reorg 任务（如 DDL 和 import）引入了全局排序，大幅减少了存储中额外资源的消耗。用户可以使用外部存储简化操作并节省成本
+             <b>TiKV 数据缓存</b><br />
+            TiKV 在内存中维护数据的最近版本，减少对多版本数据的反复扫描，进而提升性能。
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:32]: LGTM

File: tidb-roadmap.md [hunk 5]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+          </li>
+          <br />
+          <li>
+             <b>统计信息收集自动并行度</b><br />
+            TiDB 根据部署的节点数以及硬件规格自动设置统计信息收集的任务并行度和扫描并发度，提升收集速度。
+          </li>
+          <br />
+          <li>
+             <b>加速数据库恢复</b><br />
+            降低全量数据库恢复和 PiTR 所消耗的时间。
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:46]: LGTM

File: tidb-roadmap.md [hunk 6]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+          <br />
+          <li>
+            <b>减少统计信息收集时的 I/O 消耗</b><br />
+            当抽取部分数据样本做统计信息收集时，TiKV上只扫描样本，以减少统计信息收集所消耗的时间和资源。
           </li>
           <br />
           <li>
-            <b>通过分布式并行执行框架实现动态节点扩缩容</b><br />
-            动态调整节点分配，以满足后台任务的资源成本，同时保持稳定性和性能预期
+            <b>消除下推 LIMIT 的已知限制</b><br />
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:73]: LGTM

File: tidb-roadmap.md [hunk 7]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+          </li>
+          <br />
+          <li>
+            <b>Cascades Optimizer </b><br />
+            引入更成熟强大的优化器框架，扩展当前优化的基础能力。
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:78]: LGTM

File: tidb-roadmap.md [hunk 8]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
           </li>
           <br />
           <li>
-            <b>通过分布式并行执行框架实现动态节点扩缩容</b><br />
-            动态调整节点分配，以满足后台任务的资源成本，同时保持稳定性和性能预期
+            <b>消除下推 LIMIT 的已知限制</b><br />
+          </li>
+          <br />
+          <li>
+            <b>Cascades Optimizer </b><br />
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:77]: "优化器" -> "优化器框架"

File: tidb-roadmap.md [hunk 9]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
           <li>
-            <b>移除事务大小的限制</b>
+            <b>表级别的负载均衡</b><br />
+            PD 根据每个表上各 region 的负载决定数据的调度策略。
           </li>
           <li>
-            <b>联邦查询</b>
-            TiDB 查询 planner 支持 HTAP 场景中多个存储引擎
+            <b>提升大数据量系统表的处理性能</b><br />
+            当系统表中存有大量数据时，提升查询系统表的查询性能，降低查询开销。
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:99]: LGTM

File: tidb-roadmap.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:77]:
```suggestion
            <b>Cascades 优化器框架 </b><br />
```

File: tidb-roadmap.md
Comment (Oreoxmt) [inline tidb-roadmap.md:1]: Preview: https://pingcap-docsite-preview-git-preview-pi-86d6db-oreoxmts-projects.vercel.app/zh/tidb/dev/tidb-roadmap

**Review by Oreoxmt — APPROVED** [review]

**Review by joey-yez — APPROVED** [review]

File: tidb-roadmap.md [hunk 10]
```diff
@@ -150,79 +265,64 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
-- 最快的在线添加索引 `ADD INDEX` 操作 (GA)
-- TiFlash 延迟物化 (GA)
-- TiFlash 支持数据落盘 (GA)
-- LDAP 身份认证
-- SQL 审计日志增强（仅企业版可用）
-- Partitioned Raft KV 存储引擎（实验特性）
-- 通用的会话级别执行计划缓存（实验特性）
-- TiCDC 支持以 Kafka 为下游的分布式表级别数据同步（实验特性）
-
-## 已发布版本
```
Comment (Oreoxmt) [inline tidb-roadmap.md:327]:
```suggestion
```

File: tidb-roadmap.md [hunk 11]
```diff
@@ -150,79 +265,63 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+          <br />
           <li>
             <b>增强客户端加密</b>
+            <br />支持客户端对关键字段进行加密，增强数据安全性。
           </li>
           <br />
           <li>
-            <b>增强数据脱敏</b>
+            <b>业务数据动态脱敏</b>
+            <br />基于不同数据应用场景的数据脱敏，保证重要领域的数据安全。
```
Comment (qiancai) [inline tidb-roadmap.md:319]: 不确定这一条是否和前面的 “基于标记的日志脱敏” 有重复

File: tidb-roadmap.md [hunk 12]
```diff
@@ -150,79 +265,63 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
-- [TiDB 7.6.0 Release Notes](/releases/release-7.6.0.md)
-- [TiDB 7.5.0 Release Notes](/releases/release-7.5.0.md)
-- [TiDB 7.4.0 Release Notes](/releases/release-7.4.0.md)
-- [TiDB 7.3.0 Release Notes](/releases/release-7.3.0.md)
-- [TiDB 7.2.0 Release Notes](/releases/release-7.2.0.md)
-- [TiDB 7.1.0 Release Notes](/releases/release-7.1.0.md)
-- [TiDB 7.0.0 Release Notes](/releases/release-7.0.0.md)
-- [TiDB 6.6.0 Release Notes](/releases/release-6.6.0.md)
-- [TiDB 6.5.0 Release Notes](/releases/release-6.5.0.md)
+上述表格中并未列出所有计划发布的内容。另外，不同的服务订阅版本中的功能可能有所不同。
```
Comment (qiancai) [inline tidb-roadmap.md:327]:
```suggestion
> **注意：**
>
> 上述表格中并未列出所有计划发布的内容。另外，不同的服务订阅版本中的功能可能有所不同。
```

File: tidb-roadmap.md (hunk 11 above)
Comment (ljun0712) [reply tidb-roadmap.md:319]: 不重复，日志脱敏是针对日志中的敏感数据。数据动态脱敏指的是SQL访问表数据时对特定列进行脱敏。

File: tidb-roadmap.md [hunk 13]
```diff
@@ -150,79 +265,63 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
       <td>
         <ul>
           <li>
-            <b>通过 Azure Key Vault 进行密钥管理</b>
-            <br />由 Azure Key Vault 管理的静态加密
+            <b>Google Cloud KMS</b>
+            <br />静态加密支持基于 Google Cloud KMS 的密钥管理机制。
```
[the remark this answers is not in the document]
Comment (ljun0712) [reply tidb-roadmap.md:269]: 计划完善功能，转为GA特性。如果觉得不合适，可以调整

File: tidb-roadmap.md [hunk 14]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
       <td>
         <ul>
           <li>
-             <b>分布式并行执行框架</b><br />
-            TiDB v7.2.0 引入了用于后台任务（如 DDL 和 analyze）的分布式并行执行框架，为实现这些任务在计算节点间并行化提供了基础。v7.4.0 为分布式 reorg 任务（如 DDL 和 import）引入了全局排序，大幅减少了存储中额外资源的消耗。用户可以使用外部存储简化操作并节省成本
+             <b>TiKV 数据缓存</b><br />
+            TiKV 在内存中维护数据的最近版本，以减少对多版本数据的重复扫描，进而提升性能。
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:32]: 这里是指最近的几个版本，不一定是最新的一个。

File: tidb-roadmap.md [hunk 15]
```diff
@@ -104,39 +187,71 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+            <br />跟踪单条 SQL 语句在整个生命周期中的时间消耗，包括 TiDB、TiKV、PD 和 TiFlash。
           </li>
         </ul>
       </td>
       <td>
         <ul>
           <li>
-            <b>物化视图</b>
-            <br />存储预计算结果作为持久化数据视图，以提升查询性能
+            <b>Workload Analysis</b>
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:245]:
```suggestion
            <b>负载分析</b>
```

**Review by ljun0712 — APPROVED** [review]

File: tidb-roadmap.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:269]:
```suggestion
            <br />完善静态加密基于 Google Cloud KMS 的密钥管理机制，使其成为正式功能。
```
@qiancai 这样如何？

[the remark this answers is not in the document]
Comment (qiancai) [reply tidb-roadmap.md:269]: LGTM

**Review by Frank945946 — APPROVED** [review]

File: tidb-roadmap.md [hunk 16]
```diff
@@ -28,33 +28,75 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+          <br />
+          <li>
+            <b>减少统计信息收集时的 I/O 消耗</b><br />
+            在进行统计信息收集时，可以选择在 TiKV 上仅扫描部分数据样本，以减少统计信息收集所消耗的时间和资源。
           </li>
           <br />
           <li>
-            <b>通过分布式并行执行框架实现动态节点扩缩容</b><br />
-            动态调整节点分配，以满足后台任务的资源成本，同时保持稳定性和性能预期
+            <b>移除下推 LIMIT 的已知限制</b><br />
```
Comment (Oreoxmt) [inline tidb-roadmap.md:73]:
```suggestion
            <b>移除将 Limit 算子下推到 TiKV 的已知限制</b><br />
```

File: tidb-roadmap.md [hunk 17]
```diff
@@ -104,39 +187,71 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
       <td>
         <ul>
           <li>
-            <b>TiCDC 与数据仓库或数据湖系统的集成</b>
-            <br />
+            <b>可靠的终止操作</b><br />
+            正在运行中的操作能够被立即终止，并从 TiDB 和 TiKV 中释放相应的资源。
```
Comment (Oreoxmt) [inline tidb-roadmap.md:191]:
```suggestion
            <b>可靠的查询终止</b><br />
            正在运行中的 SQL 语句能够被立即终止，并从 TiDB 和 TiKV 中释放相应的资源。
```

File: tidb-roadmap.md [hunk 18]
```diff
@@ -104,39 +187,71 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+            <br />分析 Workload Repository 中的过往负载数据，根据分析结果提出优化建议，例如 SQL 调优和统计信息收集策略调整。
           </li>
           <br />
           <li>
-            <b>支持迁移异构数据库</b>
+            <b>支持修改主键</b>
+          </li>
+          <br />
+          <li>
+            <b>支持导出 SQL 语句</b>
```
Comment (Oreoxmt) [inline tidb-roadmap.md:254]:
```suggestion
            <b>支持将数据导出为 SQL 语句</b>
```

File: tidb-roadmap.md [hunk 19]
```diff
@@ -104,39 +187,71 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
           <br />
           <li>
-            <b>TiDB 节点标签</b>
-            <br />将 DDL 操作分配到现有的或新添加的 TiDB 节点，以便将 DDL 任务与在线流量使用的计算资源隔离
+            <b>切换资源组的权限控制</b>
+            <br />只有被授予特定权限的用户才能切换自身的资源组，防止资源被滥用。
+          </li>
+          <br />
+          <li>
+            <b>显示表或 SQL 与热点 Region 的关系</b>
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:200]: 这里其实还没有确定，大概率是在系统表里。 所以可以先模糊过去。

File: tidb-roadmap.md [hunk 20]
```diff
@@ -104,39 +187,71 @@ TiDB 路线图展示了 TiDB 未来的计划。随着我们发布长期稳定版
...
+            <br />分析 Workload Repository 中的过往负载数据，根据分析结果提出优化建议，例如 SQL 调优和统计信息收集策略调整。
           </li>
           <br />
           <li>
-            <b>支持迁移异构数据库</b>
+            <b>支持修改主键</b>
+          </li>
+          <br />
+          <li>
+            <b>支持导出 SQL 语句</b>
```
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:254]:
```suggestion
            <b>支持将数据导出为 SQL 语句</b>
```

File: tidb-roadmap.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:200]:
```suggestion
            <b>挖掘热点 Region 所关联的表或 SQL</b>
```
这样好些？

[the remark this answers is not in the document]
Comment (qiancai) [reply tidb-roadmap.md:200]: 或者"支持查看表或 SQL 与热点 Region 的关系"？

[the remark this answers is not in the document]
Comment (songrijie) [reply tidb-roadmap.md:200]:
```suggestion
            <b>支持查看表或 SQL 与热点 Region 的关系</b>
```

**Review by qiancai — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**